matrix/basic_matrix.py

```diff
@@ -144,15 +151,51 @@ def diagonal(self):
             result.append(self.ls_entries[i][i])
         return result
 
+    def LU_decomp(self):
+        """
+        Returns: the LU decomposition of the self.ls_entries matrix
+            L, U where [A] = [L][U]
+            L: <Matrix> m x m lower triangular matrix
+            U: <Matrix> upper triangular matrix
+        """
+        L = self.identity(self.len_row).ls_entries  # L is square matrix
+        U = [[0] * self.len_col for x in range(self.len_row)]
+        A = copy.deepcopy(self.ls_entries)
+
+        # Need to pivot first for stability
+        for i in range(self.len_row):
+            max_row_index = np.argmax(abs(np.asarray(A)[i:self.len_row, i])) + i #todo check with Emil if use of np.asarray is allowed
```

Comment:
don't know if this is allowed, uses np.array

Reply:
Hmm interesting, while I'm not against using numpy here, I do think it will be harder to justify using some functions of numpy compared to others. 
Another possible way of implementing this just using list could be: 
```
minor_col_i = [abs(x[i]) for k,x in enumerate(A) if k >= i]
max_row_index  = minor_col_i.index(max(minor_col_i)) + i
```
Though there may be better ways as well.